addons/html_builder/static/src/builder/snippets_menu.js

```diff
@@ -112,8 +126,19 @@ export class SnippetsMenu extends Component {
         this.props.closeEditor();
     }
 
-    save() {
-        console.log("todo");
+    async save() {
+        this.isSaving = true;
+        const editableEl = this.editor.editable;
+        const saveProms = [...editableEl.querySelectorAll(".o_dirty")].map(async (dirtyEl) => {
+            await this.editor.shared.media.savePendingImages(dirtyEl);
+            const cleanedEl = this.editor.shared.dirty.handleDirtyElement(dirtyEl);
+            await this.saveView(cleanedEl);
+        });
+        const hasChange = !!saveProms.length;
+        if (hasChange) {
+            await Promise.all(saveProms);
+        }
+        browser.location.reload();
```

Comment:
You don t need this code. Or I miss something (I use it in html field for the urgent save)